### ISSUE TITLE
Why huge difference (increase) in WER after using Transf-LM compared to viterbi?

### ISSUE THREAD

**Issue by ankurdhuriya** [pytorch/fairseq#2919]
#### What is your question?
I have a question regarding the abrupt increase in wer during Inference on LibriSpeech test clean data using  transformer_lm. 
I have picked checkpoints from https://github.com/pytorch/fairseq/blob/master/examples/language_model/README.md

**WER on Libri-test-clean without transf lm (viterbi) -> 2.1530
WER on Libri-test-clean with transf lm (transformer_lm.wiki103.adaptive) -> 89.999**

**How to resolve this?**

- Data used - test librispeech clean
- Finetune checkpoint - Wav2Vec 2.0 Large (LV-60)* 960hrs
- Transformer checkpoint tried - transformer_lm.wiki103.adaptive, transformer_lm.wmt19.en

#### What have you tried?

this is how I ran both the inference ->
python examples/speech_recognition/infer.py libri-test --task audio_pretraining --nbest 1 --path english_pretrained_large_checkpoint/wav2vec2_vox_960h_new.pt --gen-subset test --results-path libri/transcript --w2l-decoder viterbi --lm-weight 2 --word-score -1 --sil-weight 0 --criterion ctc --labels ltr --max-tokens 640000 --post-process letter

python examples/speech_recognition/infer.py libri-test --task audio_pretraining --nbest 1 --path english_pretrained_large_checkpoint/wav2vec2_vox_960h_new.pt --gen-subset test --results-path libri/transcript --w2l-decoder fairseqlm --lm-model adaptive_lm_wiki103.v2/model.pt --lm-weight 2 --word-score -1 --sil-weight 0 --criterion ctc --labels ltr --max-tokens 640000 --post-process letter

#### What's your environment?

 - fairseq Version (e.g., 1.0 or master): 81677d75
 - PyTorch Version (e.g., 1.0) 3.7.9
 - OS (e.g., Linux): Linux

**Comment by alexeib** [comment]
ive never tried that language model with librispeech. it has a different lexicon, domain, etc. also you need to upper case the entire dict if you want htis to have a chance of working, because the ctc model outputs only capital letters

**Comment by harveenchadha** [comment]
@alexeib Can you please let me know which is the transformer model that is tried with wav2vec to get outputs as mentioned in the paper?

**Comment by alexeib** [comment]
this is the model:
https://dl.fbaipublicfiles.com/wav2letter/sota/2019/lm/lm_librispeech_word_transformer.pt

and this is the dictionary which you should upper case after downloading:
https://dl.fbaipublicfiles.com/wav2letter/sota/2019/lm/lm_librispeech_word_transformer.dict

both of those are from https://github.com/facebookresearch/wav2letter/tree/master/recipes/sota/2019 which is linked in the wav2vec readme

**Comment by LamyaMohamed** [comment]
any update?
